src/ServiceControl/Operations/ErrorIngestion.cs

```diff
@@ -92,7 +92,7 @@ protected override async Task ExecuteAsync(CancellationToken stoppingToken)
                         batchSizeMeter.Mark(contexts.Count);
                         using (batchDurationMeter.Measure())
                         {
-                            await ingestor.Ingest(contexts);
+                            await ingestor.Ingest(contexts, stoppingToken);
                         }
                     }
                     catch (OperationCanceledException) when (stoppingToken.IsCancellationRequested)
```

Comment:
Throwing here will clear the messsage contexts without setting them to completed or failed and leave the OnMessage calls stuck. Is that a problem?

Reply:
Changes have been heavily refactored since comments and code has been modified to ensure task are completed as cancelled